tests/metrics_tests.cpp

```diff
@@ -0,0 +1,718 @@
+
+using namespace mesos::internal::tests;
+
+using mesos::modules::ModuleManager;
+
+using mesos::modules::metrics::ContainerStartRequest;
+using mesos::modules::metrics::ContainerStartResponse;
+using mesos::modules::metrics::LegacyState;
+
+using mesos::slave::Isolator;
+
+using mesosphere::dcos::metrics::MetricsIsolator;
+using mesosphere::dcos::metrics::parse;
+
+using process::Clock;
+using process::Future;
+using process::Owned;
+
+using std::string;
+using std::vector;
+
+using testing::DoAll;
+using testing::Return;
+
+namespace mesos {
+namespace metrics {
+namespace tests {
+
+const string METRICS_PROCESS = "metrics-service";
+const string API_PATH = "container";
+const string LEGACY_HOST = "metricshost";
+const Duration REQUEST_TIMEOUT = Seconds(5);
+
+// A map of container IDs and their associated ports for testing.
+
+
+class MetricsTest : public MesosTest
+{
+protected:
+  virtual void SetUp()
+  {
+    MesosTest::SetUp();
+
+    // The metrics module will only set its legacy state path data member if the
+    // directory exists at the time the module is loaded, so we create it now.
+    statePath = path::join(os::getcwd(), "state");
+    os::mkdir(path::join(statePath, "containers"));
+
+    // Construct the metrics module configuration.
+    const string ip = stringify(process::address().ip);
+    const string port = stringify(process::address().port);
+    const string config =
+"{\n"
+"  \"libraries\": [ {\n"
+"    \"file\": \"" + path::join(MODULES_BUILD_DIR,
+                                ".libs",
+                                "libmetrics-module.so") + "\",\n"
+"    \"modules\": [ {\n"
+"      \"name\": \"com_mesosphere_dcos_MetricsIsolatorModule\",\n"
+"      \"parameters\": [\n"
+"        {\"key\": \"dcos_metrics_service_scheme\", \"value\": \"http\"},\n"
+"        {\"key\": \"dcos_metrics_service_network\", \"value\": \"inet\"},\n"
+"        {\"key\": \"dcos_metrics_service_address\",\n"
+"         \"value\": \"" + ip + ":" + port + "\"},\n"
+"        {\"key\": \"dcos_metrics_service_endpoint\",\n"
+"         \"value\": \"/" + METRICS_PROCESS + "/" + API_PATH + "\"},\n"
+"        {\"key\": \"legacy_state_path_dir\",\n"
+"         \"value\": \"" + statePath + "\"},\n"
+"        {\"key\": \"request_timeout\",\n"
+"         \"value\": \"" + stringify(REQUEST_TIMEOUT) + "\"}\n"
+"      ]\n"
+"    } ]\n"
+"  } ]\n"
+"}";
+
+    Try<JSON::Object> json = JSON::parse<JSON::Object>(config);
+    ASSERT_SOME(json);
+
+    Try<Modules> _modules = protobuf::parse<Modules>(json.get());
+    ASSERT_SOME(_modules);
+
+    modules = _modules.get();
+
+    // Initialize the metrics module.
+    Try<Nothing> result = ModuleManager::load(modules);
+    ASSERT_SOME(result);
+
+    // Create an instance of the module and store it for testing.
+    Try<Isolator*> isolator_ =
+      ModuleManager::create<Isolator>(
+          "com_mesosphere_dcos_MetricsIsolatorModule");
+
+    ASSERT_SOME(isolator_);
+    isolator = isolator_.get();
```

Comment:
Is this isolator pointer deallocated anywhere?

Reply:
Nope! I added a `delete` in the test teardown, thanks!